hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/calculator/ConsensusCreateTopicFeeCalculator.java

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: Apache-2.0
+package com.hedera.node.app.service.consensus.impl.calculator;
+
+import static org.hiero.hapi.fees.FeeScheduleUtils.lookupServiceFee;
+
+import com.hedera.hapi.node.base.HederaFunctionality;
+import com.hedera.hapi.node.transaction.TransactionBody;
+import com.hedera.node.app.spi.fees.CalculatorState;
+import com.hedera.node.app.spi.fees.ServiceFeeCalculator;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import org.hiero.hapi.fees.FeeResult;
+import org.hiero.hapi.support.fees.Extra;
+import org.hiero.hapi.support.fees.FeeSchedule;
+import org.hiero.hapi.support.fees.ServiceFeeDefinition;
+
+public class ConsensusCreateTopicFeeCalculator implements ServiceFeeCalculator {
+
+    @Override
+    public void accumulateServiceFee(
+            @NonNull TransactionBody txnBody,
+            @Nullable CalculatorState calculatorState,
+            @NonNull FeeResult feeResult,
+            @NonNull FeeSchedule feeSchedule) {
+        long keys = 0;
+        final var op = txnBody.consensusCreateTopicOrThrow();
+        if (op.hasAdminKey()) {
+            keys += 1;
```

Comment:
We should count number of simple keys in each and add those instead of 1. I think we are using `countKeys(key)` method

Reply:
so you mean if (op.hasAdminKey() { keys += countKeys(op.adminKey()) } ?